### PR TITLE
Finish User Profile CRUD

### LINKED ISSUE #16: Finish CRUD 4 User Profile
Finish the CRUD operations for the User Profile Entity

### PR DESCRIPTION
**Pull request by twhitelow1**

### CODE REVIEW

File: src/SnowBuddies.Api/Controllers/UserProfileController.cs [hunk 1]
```diff
@@ -21,15 +25,58 @@ public UserProfileController(IUserProfileService userProfileService, IMapper map
         [ProducesResponseType(404)]
         [ProducesResponseType(400)]
         [ProducesResponseType(500)]
-        public async Task<IActionResult> GetAllUsers()
+        public async Task<IActionResult> GetAllUserProfilesAsync()
         {
-            var users = await _userProfileService.GetAllAsync();
```
Comment (VIbanichuka) [inline src/SnowBuddies.Api/Controllers/UserProfileController.cs:26]: Perhaps, the variable should be "userProfiles".
var userProfiles = await _userProfileService.GetAllAsync();

File: src/SnowBuddies.Api/Controllers/UserProfileController.cs [hunk 2]
```diff
@@ -21,15 +25,58 @@ public UserProfileController(IUserProfileService userProfileService, IMapper map
...
+            var users = await _userProfileService.GetAllUserProfilesAsync();
```
Comment (VIbanichuka) [inline src/SnowBuddies.Api/Controllers/UserProfileController.cs]: Perhaps, the variable should be "userProfiles".
var userProfiles = await _userProfileService.GetAllUserProfilesAsync();

File: src/SnowBuddies.Api/Controllers/UserProfileController.cs [hunk 3]
```diff
@@ -21,15 +25,58 @@ public UserProfileController(IUserProfileService userProfileService, IMapper map
...
         }
+        [HttpGet("{userProfileId}")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(204)]
+        [ProducesResponseType(404)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(500)]
+        public async Task<IActionResult> GetUserProfileById(Guid userProfileId)
+        {
+            var existingUser = await _userProfileService.GetUserProfileByIdAsync(userProfileId);
```
Comment (VIbanichuka) [inline src/SnowBuddies.Api/Controllers/UserProfileController.cs]: Perhaps, the variable should be "existingUserProfile"

File: src/SnowBuddies.Api/Controllers/UserProfileController.cs (hunk 2 above)
Comment (twhitelow1) [reply src/SnowBuddies.Api/Controllers/UserProfileController.cs]: ah you are right! I gotta be more aware when im reading and copying the code. I made this looking at the UserController.

File: src/SnowBuddies.Api/Controllers/UserProfileController.cs (hunk 3 above)
Comment (twhitelow1) [reply src/SnowBuddies.Api/Controllers/UserProfileController.cs]: Yes and same thing here. I think that was ChatGPT. Good catches! this is what code review is all about thanks.

**Review by VIbanichuka — COMMENTED** [review]
Hi Todd, I have left some comments. What do you think?

File: src/SnowBuddies.Api/Controllers/UserProfileController.cs (hunk 3 above)
Comment (twhitelow1) [reply src/SnowBuddies.Api/Controllers/UserProfileController.cs]: I see a lot more and I am fixing it. I am going to review each file I created and the names.

File: src/SnowBuddies.Api/Controllers/UserProfileController.cs (hunk 2 above)
Comment (VIbanichuka) [reply src/SnowBuddies.Api/Controllers/UserProfileController.cs]: Yeah, no biggie. Please inform me when you have made a new commit. Thanks)

**Comment by twhitelow1** [comment]
> Hi Todd, I have left some comments. What do you think?

Yes I agree with all of that I see several more cases of this and I am doing a full audit on the UserProfile related files and checking the names of variables.

**Review by VIbanichuka — APPROVED** [review]
It looks good to me

**Pull request merged**